### ISSUE TITLE
Custom Field - Can't update, delete data from ticket reply

### ISSUE THREAD

**Issue by Rezorf** [freescout-helpdesk/freescout#2160]
Clead installation of the engine version 1.8.19, no errors, the cron task is running (by "www-data" user). The "User Portal" module and "Custom Fields" module are installed (also of the latest versions, installed yesterday, there were no errors during the installation). When submitting a reply from the user portal, custom fields are filled in successfuly & it works as expected. But when answering a ticket with making changes to custom fields, it does not happen (the data is not deleted, does not change). If the field is "required", then for an empty field a requirement is issued to fill it, but data also not saved saved after reply. Tried removing modules and installing them in different order - no change. Tested with input type filds & selection type fields.

I also ask you to add the "hidden" setting to the settings of additional fields, so that I could pass the data pre-filled from the URL, but the user did not see them (now ths can be solved by using CSS in the customization module, but it would be correct to get an improvement from you).

Also, the activation of custom fields in the "user portal" settings should be placed a little higher ( they primarily relate to the form, not to the widget), or at least describe this clearly in the instructions for the "user portal" module - I spent a lot of time on this to find this setting:
![177683297-239d4fcc-ba2a-47f1-9a9c-897a9b7d4025 1](https://user-images.githubusercontent.com/108843908/185051007-7f956d78-798a-4755-b39d-651854e5ed06.png)

https://user-images.githubusercontent.com/108843908/185051061-f424f775-b564-4590-9942-78ba3376c302.MP4

**Comment by Rezorf** [comment]
Just updated to 1.8.21, custom fields to 1.0.13, problem still exists...

I can give you remote access, please give email

**Comment by freescout-helpdesk** [comment]
Please make sure not to mix different topics into one issue in the future.

> Clead installation of the engine version 1.8.19, no errors, the cron task is running (by "www-data" user). The "User Portal" module and "Custom Fields" module are installed (also of the latest versions, installed yesterday, there were no errors during the installation). When submitting a reply from the user portal, custom fields are filled in successfuly & it works as expected. But when answering a ticket with making changes to custom fields, it does not happen (the data is not deleted, does not change). If the field is "required", then for an empty field a requirement is issued to fill it, but data also not saved saved after reply. Tried removing modules and installing them in different order - no change. Tested with input type filds & selection type fields.

We've managed to reproduce the issue.

> I also ask you to add the "hidden" setting to the settings of additional fields, so that I could pass the data pre-filled from the URL, but the user did not see them (now ths can be solved by using CSS in the customization module, but it would be correct to get an improvement from you).

See https://freescout.net/request-feature/ or https://github.com/freescout-helpdesk/freescout/wiki/FreeScout-Modules#5-modifying-modules

> Also, the activation of custom fields in the "user portal" settings should be placed a little higher ( they primarily relate to the form, not to the widget), or at least describe this clearly in the instructions for the "user portal" module - I spent a lot of time on this to find this setting:

This has been fixed in the latest version of End-User Portal Module.

**Comment by freescout-helpdesk** [comment]
Update Custom Fields Module to the latest version and run `php artisan freescout:module-build` console command.

**Comment by freescout-helpdesk** [comment]
Also make sure to update to the latest version Custom Mailbox Signatures Module if you have it.

**Comment by Rezorf** [comment]
> Update Custom Fields Module to the latest version and run `php artisan freescout:module-build` console command.
At first this command didn't work - there were errors about incorrect symlinks for several modules, but I think I fixed it by myself: some symlinks from the var/www/html/Modules/ folder referred to the Public folder in the internal modules folder. Some modules had another symlink linked to nowhere. I removed the symlinks in these folders and created real Public folders (empty). I do not know if this is correct or not, but the error is gone. Before that, I tried to remove the "problem" modules and install them again - it did not help (wrong symlinks also created). After creating empty Public folders for "problem" modules, executing  `php artisan freescout:module-build` command still shows some errors (I'm attaching a screenshot - I think they are not critical). Also yesterday there were several updates from you, after all this everything seems to be working as intended (custom field data is saved immediately after input). Testing further, thanks.

![2](https://user-images.githubusercontent.com/108843908/185296883-0a4346b4-6a8f-4aee-b63e-9fad122e454a.png)
